### PR TITLE
[cling] Implement value printing for `std::filesystem::path`

### LINKED ISSUE #9672: Crash when value-printing `path` objects
- [x] Checked for duplicates

Without runtime modules, there is a silent crash when cling tries to `printValue` a `path` object, and the interpreter exits with error code 11:

```
$ root -l
root [0] #include <filesystem>
root [1] auto p = std::filesystem::path("/some/path/foo.cpp");
root [2] std::cout << p << '\n';
"/some/path/foo.cpp"
root [3] p
/tmp echo $status
11
```

With runtime modules `<filesystem>` cannot be included, see https://github.com/root-project/root/issues/9670

### PR DESCRIPTION
**Pull request by devajithvs**
# This Pull request:

## Checklist:

- [ ] tested changes locally
- [ ] updated the docs (if necessary)

This PR fixes #9672

### CODE REVIEW

File: interpreter/cling/include/cling/Interpreter/RuntimePrintValue.h [hunk 1]
```diff
@@ -241,6 +251,13 @@ namespace cling {
     return collectionPrinterInternal::printValue_impl(obj);
   }
 
+#if __cplusplus >= 201703L
+  // Specialization for std::filesystem::path
+  template <> inline std::string printValue(const std::filesystem::path* obj) {
+    return collectionPrinterInternal::printValue_impl(obj);
+  }
```
Comment (hahnjo) [inline interpreter/cling/include/cling/Interpreter/RuntimePrintValue.h:258]: Do we need this indirection or can we just declare `std::string printValue(const std::filesystem::path* value)` as we do for primitive data types and `std::string` etc. above?

Comment (dpiparo) [reply interpreter/cling/include/cling/Interpreter/RuntimePrintValue.h:258]: I think we could live without unless there is a reason that I missed.

Comment (vepadulano) [reply interpreter/cling/include/cling/Interpreter/RuntimePrintValue.h:258]: I agree, the simple overload should be enough

**Review by dpiparo — APPROVED** [review]
LGTM!

File: interpreter/cling/include/cling/Interpreter/RuntimePrintValue.h [hunk 2]
```diff
@@ -232,6 +235,13 @@ namespace cling {
       }
       return str + " }";
     }
+
+#if __cplusplus >= 201703L
+    // For std::filesystem::path
+    inline std::string printValue_impl(const std::filesystem::path* obj) {
```
Comment (hahnjo) [inline interpreter/cling/include/cling/Interpreter/RuntimePrintValue.h:241]: Why do we have to go through `printValue_impl`, why isn't `printValue` enough here (as we do for `std::string`, `std::source_location` and others)?

Comment (devajithvs) [reply interpreter/cling/include/cling/Interpreter/RuntimePrintValue.h]: My bad, it should've been `printValue`. Done!

**Review by hahnjo — APPROVED** [review]

**Pull request merged**